# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/WRHitTester.cpp
# Commit: cf96a8764a2a
# Full Hash: cf96a8764a2afa1d928319ca451dbd43d0fc53fe
# Author: Dan Robertson <drobertson@mozilla.com>
# Date: 2022-10-11 03:54:04
# Description:
#   Bug 1793406 - Guard against a null pointer deref if no APZC was found. r=botond
#   
#   If no target APZC was found, guard agains a nullptr deref, and do not check the
#   fixed position side bits of the hit.
#   
# ==============================================================================

diff -r feac609f7fe0 -r cf96a8764a2a gfx/layers/apz/src/WRHitTester.cpp
--- a/gfx/layers/apz/src/WRHitTester.cpp	Mon Oct 10 18:16:29 2022 +0000
+++ b/gfx/layers/apz/src/WRHitTester.cpp	Mon Oct 10 18:37:14 2022 +0000
@@ -228,7 +228,7 @@
       // match what is found in the hit-testing tree node.
       MOZ_ASSERT(sideBits == hit.mNode->GetFixedPosSides(),
                  "Fixed position side bits do not match");
-    } else if (hit.mTargetApzc->IsRootContent()) {
+    } else if (hit.mTargetApzc && hit.mTargetApzc->IsRootContent()) {
       // If the hit element is not a fixed position element, then the hit test
       // result item's side bits should not be populated.
       MOZ_ASSERT(sideBits == SideBits::eNone,
